main.py

```diff
@@ -131,8 +131,10 @@ def train(args, lm, dataset, tp, dp, saving_folder):
         # device, else it runs all of main.py n_devices times (????).
         # presumably its for multi-gpu training but i haven't learned how yet
         max_epochs=tp.epochs, val_check_interval=tp.val_check_epoch_frac)
-
-    mytrainer = Trainer(lm, tp, start_time=start_time)
+    
+    mytrainer = Trainer(lm, tp, 
+                        expected_batches_per_epoch=len(dataset.train_dataloader(tp.batch_size)), 
```

Comment:
this code assumes dataset.train_dataloader will return a dataloader of the same length every time you call it - in particular, that the length here will be the same length as the actual dataloader sent for training (generated below, line 240). that happens to be true but i dont like assuming.
to remove the assumption, create the train dataloader once (e.g. in line 134, with `tdl = dataset.train_dataloader(tp.batch_size)`), and then use the same one each time (i.e. in the initiation of mytrainer, and in the call to pltrainer.fit ).